cabinets/plugins.py

```diff
@@ -0,0 +1,75 @@
+import importlib
+import importlib.util
+import os
+import pkgutil
+import sys
+import inspect
+
+import cabinets.cabinet
+import cabinets.parser
+from cabinets.logger import info
+
+
+class CabinetsPluginError(Exception):
+    pass
+
+
+def discover(path, prefix=''):
+    plugins = pkgutil.iter_modules(path, prefix)
+    modules = set()
+    for _, name, _ in plugins:
+        module = importlib.import_module(name)
+        modules.add(module)
+    return modules
+
+
+def load_protocols(cls, protocols: dict):
+    if not cls._protocols:
+        raise CabinetsPluginError(f'No extensions registered to \'{cls.__name__}\'')
+    for protocol in cls._protocols:
+        if protocol in protocols:
+            raise CabinetsPluginError(f'Extension \'{protocol}\' already registered '
+                                      f'to {protocols[protocol].__qualname__}')
+        protocols[protocol] = cls
+    info(f"Loaded {cabinets.Parser.__name__} plugin '{cls.__name__}'")
+
+
+# TODO: could be combined with `load_protocols` fairly easily
```

Comment:
Yeah I like the idea of keeping them separate so they _could_ be used separately if desired, but maybe a more generic plugin loading function makes sense? It would probably make the logging less specific, but assuming we put class information in the message it should be fine.

Reply:
Yeah i think combining to a more generic one is the move.

Reply:
Do you want to do that now or file a separate issue?